### DISCUSSION TITLE
Intersect `MultiPolygon` with `Polygon` (without dissolving faces)

### DISCUSSION THREAD

**Discussion by micycle1** [locationtech/jts#898] [Q&A] [answered]
What is the best approach to preserve the faces of a multipolygon when it is intersected with a polygon?

One approach of course is to intersect each polygon of the multipolygon with the target individually, but is there a more appropriate/faster way?

### MultiPolygon...
![1658252813017](https://user-images.githubusercontent.com/9304234/179816405-aec8bbfd-746a-4c11-93f5-f4e414208b47.png)

### ...intersected with:

![1658252781711](https://user-images.githubusercontent.com/9304234/179816461-729b390f-1421-4b23-b811-3828091b5746.png)

### Desired output

![1658252709064](https://user-images.githubusercontent.com/9304234/179816516-ea04c3d2-b245-4296-a185-3af3c4b03534.png)

### Actual output

![1658252781711](https://user-images.githubusercontent.com/9304234/179816535-8a146e55-24cc-4d6b-bd7b-62e9fe4b88b8.png)

**Comment by dr-jts** [comment, marked as answer]
Actually that doesn't look like a valid MultiPolygon, since the faces are edge-adjacent.  It should be represented as a GeometryCollection, or a an array or List of Polygons.

The usual process for doing this is called Polygon Coverage Overlay.  JTS doesn't currently have a class that does this (although I hope to implement one medium-term).  But it can be constructed by using noding and polygonization.  There is a simple implementation [here](https://github.com/locationtech/jts/blob/master/modules/app/src/main/java/org/locationtech/jtstest/function/PolygonOverlayFunctions.java#L79).  If you want to keep only the resultants in the polygon B, you can use Point-In-Polygon tests to do that.

Performing overlay as a bulk process should be faster than performing a series of individual intersection operations, primarily due to the lower overhead of noding the linework in a single step, rather than many separate ones.
